automation/devops_automation_infra/utils/node_daemon.py

```diff
@@ -0,0 +1,59 @@
+import json
+import logging
+import os
+
+from automation_infra.utils import waiter, timer
+from devops_automation_infra.utils import consul
+from infra.model import cluster_plugins
+from pipeng.services import pipe
+FACE_FLOW_TYPE = 1
+BODY_FLOW_TYPE = 2
+SERVICE_ADDRESS_SEPARATOR = ":"
+OPEN_VINO_CONSUL_PATH = "supervisor_local-agent-default/pipe_defs:OV"
+TRT_CONSUL_PATH = "supervisor_local-agent-default/pipe_defs:TRT"
+THIS_DIRNAME = os.path.dirname(os.path.realpath(__file__))
+ZMQ_KEYS_DIR = os.path.realpath("%s/../../../zmq/certificates" % THIS_DIRNAME)
+from pipeng.utils import tracks
+from devops_automation_infra.k8s_plugins.consul import Consul
+from devops_automation_infra.k8s_plugins.kafka import Kafka
+
+CONSUL_NAME = 'pipeng_local-agent-default'
+
+
+def create_kafka_tracks_reader(cluster, reader_offset='latest', topic="anv.tracks.pipeng.new-tracks"):
+    kafka_client = cluster.Kafka
+    tracks_reader = tracks.Reader(kafka_client, topic=topic, reader_offset=reader_offset)
+    return tracks_reader
+
+
+def list_pipes_from_consul(cluster):
+    consul_client = cluster.Consul.create_client()
+    instances = consul.get_service_nodes(consul_client, "pipeng")
+    logging.debug(f'Found {instances} in consul')
+    return instances
+
+
+def get_consul_path(layer, field, node_id=None):
+    if node_id is None:
+        path = f"{layer}/{CONSUL_NAME}/{field}"
+    else:
+        path = f"{layer}/{node_id}/{field}"
+    return path
```

Comment:
Not so important buy you could change it to a cleaner look (no need to define path):
```
if node_id is None:
    return ....
return f"{layer}/{node_id}/{field}"
```
Or even in a one liner (if you like)

Reply:
great upgrade